### PR TITLE
Add arrow/1.0.0 recipe

### LINKED ISSUE #63: [request] Add appache arrow library
I'm requesting the apache arrow library version 0.14.1 to be added:

Link to the website: https://arrow.apache.org/docs/index.html
There is a slightly outdated recipe here: https://github.com/FiligreeTech/conan-arrow

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **arrow/0.16.0**

Depends on #1179
Depends on #1180
Depends on #1181
Depends on #1146

Fixes #63

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Croydon** [comment]
_madebr creates pull request which depends on 4 other open pull requests_

You are on fire 🔥 😄

File: recipes/arrow/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.arch != "x86_64":
+            del self.options.with_sse42
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if not self.settings.build_type:
+            self.settings.build_type = "Release"
```
Comment (uilianries) [inline recipes/arrow/all/conanfile.py]:
```suggestion
```
do not force settings

**Comment by Croydon** [comment]
Please put `Fixed #63` to PR, thanks!

**Comment by htfy96** [comment]
Hi, thanks for your contribution on adding this recipe! Just several suggestions:

1. conda-forge already had an arrow-cpp recipe at https://github.com/conda-forge/arrow-cpp-feedstock/blob/68f4020a23e13039cab49927b6e9e51f27fc33ab/recipe/meta.yaml
1. Arrow supports [modular builds for some of its components](https://github.com/apache/arrow/blob/master/docs/source/developers/cpp/building.rst#modular-build-targets). I was wondering if it's possible to break this into four libraries (arrow, parquet, gandiva and plasma)
2. libparquet.so isn't included in current recipe. Maybe we can add it if `options.with_parquet == True`?

**Comment by madebr** [comment]
>     2. Arrow supports [modular builds for some of its components](https://github.com/apache/arrow/blob/master/docs/source/developers/cpp/building.rst#modular-build-targets). I was wondering if it's possible to break this into four libraries (arrow, parquet, gandiva and plasma)

I reviewed the options and added these as options and as a component.
Only arrow and plasma can be built atm, because of missing dependencies of parquet and gandiva.

**Comment by Croydon** [comment]
Nit: Please update PR title to reflect that 1.0.0 gets added not 0.16.0

File: recipes/arrow/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+        self.info.options.with_re2 = self._with_re2()
+        self.info.options.with_jemalloc = self._with_jemalloc()
+        self.info.options.with_openssl = self._with_openssl()
+        self.info.options.with_boost = self._with_boost()
+        self.info.options.with_glog = self._with_glog()
+        self.info.options.with_grpc = self._with_grpc()
+
+    def package_info(self):
+        self.cpp_info.components["libarrow"].libs = [self._lib_name("arrow")]
+        self.cpp_info.components["libarrow"].filenames["cmake_find_package"] = "Arrow"
```
Comment (SpaceIm) [inline recipes/arrow/all/conanfile.py:383]: too late to review, but does it create one config file per component?

Comment (madebr) [reply recipes/arrow/all/conanfile.py:383]: I just checked, it creates `FindArrow.cmake`, `FindPlasma.cmake`, `FindParquet.cmake` .. config files with a lower case `arrow`/`plasma`/`parquet` target (or `arrow_static`/`plasma_static`/`parquet_static` if built statically).

Comment (jgsogo) [reply recipes/arrow/all/conanfile.py:383]: CMake generators create only one file: `FindArrow.cmake` that provide the three targets, PkgConfig generator creates three files that follow the names of the components: `arrow.pc`, `plasma.pc` and `parquet.pc`.

Check PR here, there are some issues in the recipe: https://github.com/conan-io/conan-center-index/pull/2592

**Pull request merged**